compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -415,6 +415,9 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 let result = self.raw_eq_intrinsic(&args[0], &args[1])?;
                 self.write_scalar(result, dest)?;
             }
+            sym::typed_swap => {
+                self.typed_swap_intrinsic(&args[0], &args[1])?;
+            }
```

Comment:
That's not type, so we'd miss UB.

Whenever an intrinsic has a body but that body has less UB than the documentation permits, we *must* overwrite it in the interpreter (at least in Miri) to avoid missing UB. This is language UB, after all. We should have some sort of check to ensure we don't forget that...